fabrics.c

```diff
@@ -320,6 +320,27 @@ static bool ctrl_matches_connectargs(char *name, struct connect_args *args)
 	cargs.trsvcid = parse_conn_arg(addr, ' ', conarg_trsvcid);
 	cargs.host_traddr = parse_conn_arg(addr, ' ', conarg_host_traddr);
 
+	if (!strcmp(cargs.subsysnqn, NVME_DISC_SUBSYS_NAME)) {
```

Comment:
I suggest moving this test further down after the `!strcmp(cargs.subsysnqn, args->subsysnqn)`. We don't need to test this if we aren't looking for a discovery controller.